### PR TITLE
console: make global console work in workers

### PR DESCRIPTION
**Pull request by addaleax**
##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/ayojs/ayo/blob/latest/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

console

### CODE REVIEW

**Review by sandfox — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by pup** [comment]
landed in d38439c!

**Comment by addaleax** [comment]
Sorry, I should have marked this as WIP – the tests are failing here because it also depends on the main workers PR

**Review by aqrln — APPROVED** [review]

File: lib/console.js [hunk 1]
```diff
@@ -245,7 +245,12 @@ Console.prototype.groupEnd = function groupEnd() {
     this[kGroupIndent].slice(0, this[kGroupIndent].length - 2);
 };
 
-module.exports = new Console(process.stdout, process.stderr);
+if (process.binding('worker').threadId === 0) {
+  module.exports = new Console(process.stdout, process.stderr);
+} else {
+  const { SyncWriteStream } = require('internal/fs');
+  module.exports = new Console(new SyncWriteStream(1), new SyncWriteStream(2));
```
Comment (TimothyGu) [inline lib/console.js]: A problem I've seen previously in other multithreaded applications is that sometimes the console output from different threads gets multiplexed like

normal:
```
message 1
message 2
```

multiplexed:

```
message message 1
2
```

Are we certain that the same would not happen here?

Comment (Qard) [reply lib/console.js]: The main thread is an async stream, so likely messages from the main thread would print well after messages from worker threads anyway. I think this most likely *would* suffer from that same problem though, given that there's no synchronization of the logging.

I think for this to work fully correctly, logging from the main thread and all worker threads might need to be coordinated through a single, separate logging thread.

Comment (TimothyGu) [reply lib/console.js]: @Qard right... either that, or just add a mutex/`Atomics.wait()` protecting any attempt to use the console.

Comment (addaleax) [reply lib/console.js]: Ugh … yeah. I’m not sure what I’d prefer yet, or how much of a real problem this is (all of these streams *are* sync under most circumstances, so I’d expect this to be a non-issue most or all of the time…)

**Review by TimothyGu — APPROVED** [review]

**Review by Qard — APPROVED** [review]
I'm okay with landing this as-is. We can think about the print timing thing later, if it turns out to be an issue.

**Pull request merged**